### ISSUE TITLE
[Survey] How are you using this package?

### ISSUE THREAD

**Issue by iBicha** [iBicha/UnityYoutubePlayer#73]
**tl;dr : share how you are using this package to prioritize feature development**

What are you using UnityYoutubePlayer for? I would like to get some feedback from the community to understand how this package is being used, and what use cases are being solved.

Here's my assumption (and assumptions are usually either incomplete or plain wrong) on some use cases:
- Playing videos for prototypes (easy to use Youtube at the beginning, and switch to use a more serious/robust solution once a project becomes more refined)
- Use for games relying on a lot of cinematic elements, or contains a lot of cutscenes
- For games to include a list of tutorials on how to play the game
- Building just a youtube video player (although since Unity is a full featured 3d game engine, I'm not sure if this is a good technical choice)

Does your use case fit in the list? Are you building something else?

I've been going at this project on and off solo, and with my limited time to work on this project for free, there's a lot of "possible" directions to go from here, so efforts should be directed towards what's useful. Some ideas:
- Supporting subtitles 
- Supporting 360 videos (currently has issues with sound)
- Supporting higher quality videos (need to play audio and video separately, and figure out webm decoding) 
  - usually I found not even worth investigating, since the video player starts lagging with higher resolution
- Support playing live streams
- Supporting syncing multiple videos (e.g. for syncing two videos where one is a texture and the other is a bump map or something - I don't know...)
- Whatever else.

**Comment by Shantibhushan95** [comment]
I'm using this for a VR media player (under-development) on a standalone Android VR headset. But, the throttling issues have made it almost unusable. Was looking about moving to the recommended yt-dlp package, found nothing so far. Also any help about muxing the video and audio streams for higher resolutions will be a godsend, because on a 4K VR headset watching 360 videos in 720p resolution is a torture.

**Comment by ubergeekseven** [comment]
@ubergeekseven I think that simple use for viewing is the best feature for me.
If we could link the executable desktop version to the code to send the request locally, I think it would be very useful. Since the main downloader project is pretty much dead and another has come about that works exactly the same way, we could just point to that executable.
I renamed the projects executable and it works exactly the same as the original. Not needing a virtual server setup for it and instead using the package as a dependency on the desktop works well for my case.

I know this doesn't solve mobile though.

**Comment by iBicha** [comment]
These are great use cases - thanks for sharing.
@ubergeekseven a bit late, but I've added support for yt-dlp, both on a server and local instance. That should help with the throttling issue.

As for the 360 videos, it used to work with just the video and half-decent quality, but now it does not work at all.

**Comment by harshithsirigeri** [comment]
Hi I am using it develop a game for toddlers with autism to learn basic skills. Using video a reinforcer, hence using your package. Thank you, helped my son already.

**Comment by KimRyuan** [comment]
I'm using the Android app to play a designated YouTube video..!  The URL from the PC demo version as Video Player can also be played on Android, so I'll save the URL and run it on Android! (I'm not a professional programmer yet)

1. input URL in PC Youtube Player Script
ex) https://youtu.be/qxdk6nyt4TM

2. Play Editor! Than It Give me "New URL" in Video Player!
ex) https://rr1---sn-ab02a0nfpgxapox-jwwe7.googlevideo.com/videoplayback?expire=1692693175&ei=Vx7kZOKcK-y12roPoeajoAo&ip=106.255.220.105&id=o-AMa1aS2VAB3EFjlECbxdQ3_QRysdONp5PL98uuozwDJ5&itag=22&source=youtube&requiressl=yes&mh=p3&mm=31%2C26&mn=sn-ab02a0nfpgxapox-jwwe7%2Csn-a5msenes&ms=au%2Conr&mv=m&mvi=1&pl=18&initcwndbps=1176250&spc=UWF9f5bYP8fDXFiH90wcJL194w3xafA&vprv=1&svpuc=1&mime=video%2Fmp4&cnr=14&ratebypass=yes&dur=240.047&lmt=1685532277797776&mt=1692671176&fvip=2&fexp=24007246%2C24363392&c=ANDROID&txp=5432434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Ccnr%2Cratebypass%2Cdur%2Clmt&sig=AOq0QJ8wRQIgOjC9enYai_YgNJZw7bJpMFrnBFF48TrK1UVOQ8OWSyMCIQCi4SjTQcp09d1mzQ7XrqJT9uTHrv7WOKWNPuRCoho_0g%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgSqJ5gHrdhVyfkUM2XpU-E5x9jyRtpBT9J2SuK55i6x4CIQD2lQxTjuBVtUQPH97nioWnZTWvttQpzeEsVT3jkPkxmQ%3D%3D

3. I save that in Script, or txt file.

4. Use that "New URLs" in Android. input URL in Video Player.
It working!

I don't know how that "New URL" comes out and how it is used, so I don't know English and programming, but.. I'm using it well!!
I made my own playlist in advance, and I'm screenshot Images or playing it on Android without going into YouTube! Thank You!

**Comment by KimRyuan** [comment]
Thank you for the additional information!!

 I used to think for a long time about why it doesn't work when the link doesn't work, but now I know why...! My own YouTube playlist app doesn't keep working, so I was thinking about what to do, but I'll have to find another way or think about it more...!

Thank you very much for making a good GitHub project!

**Comment by iBicha** [comment]
Closing, thank you everyone for the feedback.
